File: app/src/main/java/com/sortedqueue/programmercreek/adapter/ScreenSlidePagerAdapter.java
package com.sortedqueue.programmercreek.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;

import com.sortedqueue.programmercreek.fragments.SlideFragment;

import java.util.ArrayList;

/**
 * Created by Alok on 06/04/17.
 */
public class ScreenSlidePagerAdapter extends FragmentStatePagerAdapter {

    private ArrayList<Fragment> fragmentArrayList;
    public ScreenSlidePagerAdapter(FragmentManager supportFragmentManager, ArrayList<Fragment> fragmentArrayList) {
        super(supportFragmentManager);
        this.fragmentArrayList = fragmentArrayList;
    }

    @Override
    public Fragment getItem(int position) {
        return fragmentArrayList.get(position);
    }

    @Override
    public int getCount() {
        return fragmentArrayList.size();
    }

    public void addNewSlideFragment(SlideFragment slideFragment) {
        fragmentArrayList.add(slideFragment);
    }

    public void removeCurrentFragment(int currentPosition) {
        fragmentArrayList.remove(currentPosition);
    }

    @Override
    public int getItemPosition(Object object) {
        int index = fragmentArrayList.indexOf(object);

        if (index == -1)
            return POSITION_NONE;
        else
            return index;
    }
}

File: app/src/main/java/com/sortedqueue/programmercreek/adapter/ScreenSlidePagerAdapter.kt
package com.sortedqueue.programmercreek.adapter

import android.support.v4.app.Fragment
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentStatePagerAdapter
import android.support.v4.view.PagerAdapter

import com.sortedqueue.programmercreek.fragments.CreateSlideFragment

import java.util.ArrayList

/**
 * Created by Alok on 06/04/17.
 */
class ScreenSlidePagerAdapter(supportFragmentManager: FragmentManager, val allItems: ArrayList<Fragment>) : FragmentStatePagerAdapter(supportFragmentManager) {

    override fun getItem(position: Int): Fragment {
        return allItems[position]
    }

    override fun getCount(): Int {
        return allItems.size
    }

    fun addNewSlideFragment(createSlideFragment: CreateSlideFragment) {
        allItems.add(createSlideFragment)
    }

    fun removeCurrentFragment(currentPosition: Int) {
        allItems.removeAt(currentPosition)
    }

    override fun getItemPosition(`object`: Any): Int {
        val index = allItems.indexOf(`object`)

        if (index == -1)
            return PagerAdapter.POSITION_NONE
        else
            return index
    }
}
